checkov/bicep/runner.py

```diff
@@ -1,57 +1,105 @@
         "resources": resource_registry,
     }
 
+    def __init__(
+        self,
+        db_connector: NetworkxConnector = NetworkxConnector(),
+        source: str = "Bicep",
+        graph_class: Type[LocalGraph] = BicepLocalGraph,
+        graph_manager: GraphManager | None = None,
+        external_registries: list[BaseRegistry] | None = None,
+    ) -> None:
+        self.external_registries = external_registries if external_registries else []
+        self.graph_class = graph_class
+        self.graph_manager = (
+            graph_manager if graph_manager else BicepGraphManager(source=source, db_connector=db_connector)
+        )
+        self.graph_registry = get_graph_checks_registry(self.check_type)
+
+        self.definitions: dict[Path, BicepJson] = {}
+        self.definitions_raw: dict[Path, list[tuple[int, str]]] = {}
+
     def run(
         self,
-        root_folder: str | Path,
+        root_folder: str | Path | None,
         external_checks_dir: list[str] | None = None,
         files: list[str] | None = None,
         runner_filter: RunnerFilter = RunnerFilter(),
         collect_skip_comments: bool = True,
     ) -> Report:
         report = Report(Runner.check_type)
 
-        file_paths: set[Path] = set()
-        if root_folder:
-            root_path = Path(root_folder)
-            file_paths = {file_path for file_path in root_path.rglob("*.bicep")}
-        if files:
-            for file in files:
-                if file.endswith(".bicep"):
-                    file_paths.add(Path(file))
+        if self.context is None or self.definitions is None:
+            file_paths = get_scannable_file_paths(root_folder=root_folder, files=files)
+
+            self.definitions, self.definitions_raw, parsing_errors = Parser().get_files_definitions(file_paths)
+
+            report.add_parsing_errors(parsing_errors)
+
+            if external_checks_dir:
+                for directory in external_checks_dir:
+                    resource_registry.load_external_checks(directory)
+                    self.graph_registry.load_external_checks(directory)
+
+            self.context = None  # TODO: create context
+
+            logging.info("Creating Bicep graph")
+            local_graph = self.graph_manager.build_graph_from_definitions(self.definitions)
+            logging.info("Successfully created Bicep graph")
 
-        definitions, definitions_raw, parsing_errors = Parser().get_files_definitions(file_paths)
+            self.graph_manager.save_graph(local_graph)
+            self.definitions = local_graph.definitions  # TODO: adjust, when implementing variable rendering
 
-        report.add_parsing_errors(parsing_errors)
+        # run Python checks
+        self.add_python_check_results(report=report, runner_filter=runner_filter)
 
-        for file_path, definition in definitions.items():
+        # run YAML checks
+        self.add_yaml_check_results(report=report, runner_filter=runner_filter)
```

Comment:
@gruebel please call this graph checks, not YAML :)
They can be also described in JSON today, and possibly in other syntaxes in the future